pydra/engine/specs.py

```diff
@@ -50,6 +58,139 @@ def retrieve_values(self, wf, state_index=None):
         for field, value in temp_values.items():
             setattr(self, field, value)
 
+    def check_metadata(self):
+        """ checking the metadata for fields in input_spec and fields,
+            setting the default values when available and needed
+        """
+        supported_keys = [
+            "mandatory",
+            "xor",
+            "requires",
+            "default_value",
+            "position",
+            "help_string",
+            "output_file_template",
+            "output_field_name",
+            "separate_ext",
+            "argstr",
```

Comment:
shouldn't argstr only get added by shellspec? is there a need for this in base/function?

Reply:
I've moved this method to shellspec after yesterday meeting